sphinxcontrib/openapi.py

```diff
@@ -143,6 +291,11 @@ def _normalize_spec(spec, **options):
 
 def openapi2httpdomain(spec, **options):
     generators = []
+    spec_version = spec.get('openapi', spec.get('swagger', '2.0'))
+    if spec_version.startswith('2.'):
+        httpresource = _httpresource_v2
+    elif spec_version.startswith('3.'):
+        httpresource = _httpresource_v3
```

Comment:
It's possible that there will be an OpenAPI 4.0 in the future. We should have a fallthrough `else` and raise a `ValueError`